### PR TITLE
Ruby group capstone - Catalog of my things

### PR DESCRIPTION
**Pull request by Abdulwhbhussain**
[Project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/catalog_of_my_things.md#project-requirements)
[Logic](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/catalog_of_my_things.md#logic)
Each class from the [UML class diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png) should be created in a separate .rb file.
All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes should be implemented.
All properties visible in the diagram should be defined and set up in the constructor method. Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created.
All methods visible in the diagram should be implemented:
add_item method (in all classes that have that method)
should take an instance of the Item class as an input.
should add the input item to the collection of items.
should add self as a property of the item object (by using the correct setter from the item object).
can_be_archived?() in the Item class
should return true if published_date is older than 10 years.
otherwise, it should return false.
move_to_archive() in the Item class
should reuse can_be_archived?() method.
should change the archived property to true if the result of the can_be_archived?() method is true.
should do nothing if the result of the can_be_archived?() method is false.
can_be_archived?() in the Book class
should override the method from the parent class.
should return true if parent's method returns true OR if cover_state equals to "bad".
otherwise, it should return false.
can_be_archived?() in the MusicAlbum class
should override the method from the parent class.
should return true if parent's method returns true AND if on_spotify equals true.
otherwise, it should return false.
can_be_archived?() in the Movie class
should override the method from the parent class.
should return true if parent's method returns true OR if silent equals true.
otherwise, it should return false.
can_be_archived?() in the Game class
should override the method from the parent class.
should return true if parent's method returns true AND if last_played_at is older than 2 years.
otherwise, it should return false.
Add unit tests for all implemented methods.
[User interface](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/catalog_of_my_things.md#user-interface)
Create a main.rb file that will serve as your console app entry-point.
Your console app, at the start, should:
Present the user with a list of options to perform.
Let users choose an option.
If needed, ask for parameters for the option.
Have a way to quit the app.
The following options should be available:
List all books
List all music albums
List all movies
List of games
List all genres (e.g 'Comedy', 'Thriller')
List all labels (e.g. 'Gift', 'New')
List all authors (e.g. 'Stephen King')
List all sources (e.g. 'From a friend', 'Online shop')
Add a book
Add a music album
Add a movie
Add a game
All data should be preserved by saving collections in .json files.
[Database](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/catalog_of_my_things.md#database)
Create a schema.sql file with tables that will be analogical to the structure of the classes in your app. As you cannot implement inheritance in the database tables - add all properties and associations from the parent Item class as table columns to all tables based on the child classes. Thanks to that it can be used to store data in the future.
[Project documentation](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/catalog_of_my_things.md#project-documentation)
Once you have finished the development of the project, you should record a video presenting the features of the project you built. It is a video with a maximum length of 5 minutes. The content of the video should include:

A description of the project.
A demo of the project features.
You should also highlight some interesting piece of code or something you built that you are very proud of.
You all should appear in the video and talk about the project.
For recording the video you can use the Zoom recording features while in a call with your peers.

### CODE REVIEW

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Abdulwhbhussain, @MenendezON, and @Mahdi-Niazi👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
image


## To Highlight 🎉
- All tests are passed. ✔️
- Highly organized code. ✔️
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️

- [ ] I couldn't find the presentation video link in the PR description or in the README file. As requested in the project requirements, Please make sure to record a video presenting the features of the project you built. It should be a video with a maximum length of 5 minutes. The content of the video should include the following:
image

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 1]
```diff
@@ -1,2 +1,130 @@
...
+### Key Features <a name="key-features"></a>
+
+- [ ] A class with movies save the foverite moview
+- [ ] A class with book save the books
+- [ ] A class with game save the games
+- [ ] Save all of my foverite catalogs.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+🚀 Video link (https://drive.google.com/file/d/1gBlswbNlEg1XGpA8UVfsA-lKVB6sDWcv/view?usp=drive_link)
```
Comment (julie-ify) [inline README.md]: - [ ] Your video length is 7 minutes 32 seconds. Kindly note that the video should not exceed 5 minutes. Kindly provide a shorter video that meets the requirements.

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @Abdulwhbhussain, @MenendezON and @Mahdi-Niazi,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (KDlamini) [reply README.md]: Well resolved ✔️ 
Well done with the Ruby basic UI 👏🏾 💯 

image

Comment (Abdulwhbhussain) [reply README.md]: Bro Review is still Pending!?

**Review by KDlamini — APPROVED** [review]
### Hi team @Abdulwhbhussain, @MenendezON 🙋‍♀️,  

Excellent work! 💯 
You have done a great job creating a working Catalog of my Things Ruby capstone project. 🎊 🥳
You have outdone yourself in this milestone, your project is incredible!!! 🚀 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!


## Highlights 🌞

✔️  Outstanding PR title and description
✔️  Well-documented and professional README
✔️  All aggregations and associations (1-to-many relationships) are implemented
✔️  All `can_be_archived?` methods in respective classes are present
✔️  Data is preserved in JSON format on app exit and loaded when the app starts
✔️  Database schema is set up according to the UML diagram

## Video Presentation 🎥 

✔️  The recorded video is presenting the features of the project with a maximum length of 5 minutes.
✔️ The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.
✔️ All the members of the project appear in the video and talk about the project.
✔️ The video contains a highlight about a piece of the code the student finds interesting or is proud of making.

### To Improve 🔧 
To organize your files well and improve the file structure, I think it would be best if the preserved data JSON files are organized into their own directory. Having the files scattered all over the root directory makes the application look messy and disorganized.

# Status: Approved ✔️ 

oscars-standing-ovation
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**